Allow SetCipherList to operate on SSL without modifying on SSL_CTX

julek-wolfssl opened the pull request.

julek-wolfssl committed:
Allow SetCipherList to operate on SSL without modifying on SSL_CTX

SparkiDev commented on src/internal.c:
```diff
@@ -26128,25 +26129,38 @@ int SetCipherList(WOLFSSL_CTX* ctx, Suites* suites, const char* list)
         return 0;
     }
 
+    version = ctx != NULL ? ctx->method->version : ssl->version;
```
The only difference I can see when using ctx or ssl is that the version, privateKeySz and side are coming from ctx or ssl.
Extract the code that does the work into a new function and have 'haveAnon' be returned through a parameter.
Leave the SetCipherList API as it was and have it call the new function.
Add a new function SetCipherListEx that takes the ctx and ssl and uses the new function.

SparkiDev commented on wolfssl/internal.h:
```diff
@@ -2371,8 +2371,8 @@ typedef struct TLSX TLSX;
 WOLFSSL_LOCAL int  MatchSuite(WOLFSSL* ssl, Suites* peerSuites);
-WOLFSSL_LOCAL int  SetCipherList(WOLFSSL_CTX* ctx, Suites* suites,
-                                 const char* list);
+WOLFSSL_LOCAL int  SetCipherList(const WOLFSSL_CTX* ctx, const WOLFSSL* ssl,
```
When HAVE_ANON is defined, ctx or ssl is modified.

julek-wolfssl replied:
I removed setting the `haveAnon` so now they are really `const`.

julek-wolfssl committed:
Refactor haveAnon into useAnon

(ctx->|ssl->options.)useAnon means that the user has signalled that they want anonymous ciphersuites

julek-wolfssl commented:
Retest this please

julek-wolfssl committed:
Address code review
